hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockInputStream.java

```diff
@@ -574,7 +574,20 @@ private boolean shouldRetryRead(IOException cause) throws IOException {
     } catch (Exception e) {
       throw new IOException(e);
     }
-    return retryAction.action == RetryPolicy.RetryAction.RetryDecision.RETRY;
+    if (retryAction.action == RetryPolicy.RetryAction.RetryDecision.RETRY) {
+      if (retryAction.delayMillis > 0) {
+        try {
+          LOG.info("Retry read after {}ms", retryAction.delayMillis);
```

Comment:
Nit: Will this cause a LOG flooding in case of persistent error across the blocks? For example, when one DN is down.

Reply:
Changed to debug

Reply:
Thank you for the update.